test_conformance/api/test_clone_kernel.cpp

```diff
@@ -351,51 +466,467 @@ int test_clone_kernel(cl_device_id deviceID, cl_context context, cl_command_queu
     test_error( error, "clEnqueueReadBuffer failed." );
 
     // Compare the results
-    if (((int*)pbufRes)[0] != intarg)
+    test_assert_error(((int*)pbufRes)[0] == intarg,
+                      "clCloneKernel test failed. Failed to clone integer type "
+                      "argument.");
+
+    test_assert_error(
+        ((int*)pbufRes)[1] == sa.i,
+        "clCloneKernel test failed. Failed to clone structure type "
+        "argument.");
+
+    test_assert_error(
+        ((float*)pbufRes)[2] == farg,
+        "clCloneKernel test failed. Failed to clone float type argument.");
+
+    test_assert_error(
+        ((float*)pbufRes)[3] == sa.f,
+        "clCloneKernel test failed. Failed to clone structure type "
+        "argument.");
+
+    test_assert_error(
+        ((int*)pbufRes)[6] == write_val,
+        "clCloneKernel test failed.  Failed to clone cl_mem argument.");
+
+    if (bimg)
     {
-        test_error( error, "clCloneKernel test failed. Failed to clone integer type argument." );
-        return -1;
+        error = test_image_arg_shallow_clone(deviceID, context, queue,
+                                             num_elements, pbufRes, bufOut);
+        test_error(error, "image arg shallow clone test failed.");
     }
 
-    if (((int*)pbufRes)[1] != sa.i)
+    if (bdouble)
     {
-        test_error( error, "clCloneKernel test failed. Failed to clone structure type argument." );
-        return -1;
+        error = test_double_arg_clone(deviceID, context, queue, num_elements,
+                                      pbufRes, bufOut);
+        test_error(error, "double arg clone test failed.");
+    }
+
+    delete[] pbuf;
+    delete[] pbufRes;
+
+    return TEST_PASS;
+}
+
+int test_exec_enqueue_helper(cl_context context, cl_command_queue queue,
+                             BufPtr* pBuf, cl_int* svmPtr_Kernel,
+                             cl_kernel* srcKernel, cl_int* value)
```

Comment:
See above - I think we can pass most (if not all) of these by value.

Do we need to pass pBuf at all?

Minor: I'd also consider renaming this function to something like test_SVM_enqueue_helper, since this helper could work for other SVM tests and not just tests for clSetKernelExecInfo.